src/main/java/com/puppycrawl/tools/checkstyle/JavaAstVisitor.java

```diff
@@ -2160,6 +2164,28 @@ public DetailAstImpl visit(ParseTree tree) {
         return ast;
     }
 
+    /**
+     * Builds an expression node. This is used to build the root of an expression with
+     * an imaginary {@code EXPR} node.
+     *
+     * @param exprNode expression to build node for
+     * @return expression DetailAstImpl node
+     */
+    private DetailAstImpl buildExpressionNode(ParseTree exprNode) {
+        final DetailAstImpl expression = visit(exprNode);
+
+        final DetailAstImpl exprRoot;
```

Comment:
Cleaned up this code a bit, but this is basically what was in the visitExpression method.